### DISCUSSION TITLE
2FA

### DISCUSSION THREAD

**Discussion by Mtaylert** [community/community#28803] [Other Feature Feedback, Questions, & Ideas]
Hello, I recently upgraded my iphone. I no longer have access to my 2FA via the DUO app. Also, i do not have my recovery key. Can someone help me reset?

**Comment by airtower-luna** [comment]
Here's the documentation on the options you have in that situation: [Recovering your account if you lose your 2FA credentials](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/recovering-your-account-if-you-lose-your-2fa-credentials)

Considering you still seem to be logged in, first check if you can view the recovery codes again or reset TOTP. And definitely don't log out. Good luck!
